clients/cobol-lsp-vscode-extension/src/CopybooksDownloader.ts

```diff
@@ -14,81 +14,64 @@
 
 import * as fs from "fs";
 import * as vscode from "vscode";
-import { SETTINGS_SECTION } from "./constants";
-import { CopybookProfile, DownloadQueue } from "./DownloadQueue";
-import { checkWorkspace, createCopybookPath, createDatasetPath } from "./PathUtils";
+import { DEPENDENCIES_FOLDER, SETTINGS_SECTION } from "./constants";
 import { ProfileService } from "./ProfileService";
+import { DependenciesDesc, loadDepFile } from "./services/DependencyService";
+import { CopybookProfile, DownloadQueue } from "./services/DownloadQueue";
+import { checkWorkspace, createCopybookPath, createDatasetPath } from "./services/PathUtils";
 import { ZoweApi } from "./ZoweApi";
+import { CopybookResolver } from "./services/CopybookResolver";
 
 export class CopybooksDownloader implements vscode.Disposable {
     private queue: DownloadQueue = new DownloadQueue();
 
     public constructor(
+        private resolver: CopybookResolver,
         private zoweApi: ZoweApi,
         private profileService: ProfileService) { }
 
     public async redownloadDependencies(message: string = "Redownload dependencies requested.") {
-        (await vscode.workspace.findFiles(".cobdeps/**/*.dep")).forEach(dep => {
-            const errFile = dep.fsPath.substr(0, dep.fsPath.length - 4) + ".err";
-            if (fs.existsSync(errFile)) {
-                try {
-                    fs.unlinkSync(errFile);
-                } catch (error) {
-                    vscode.window.showErrorMessage(error.toString());
-                }
-            }
+        (await vscode.workspace.findFiles(DEPENDENCIES_FOLDER + "/**/*.dep")).forEach(dep => {
             this.downloadDependencies(dep, message);
         });
     }
     /**
      * @param copybooks array of copybooks names to download
      */
-    public async downloadDependencies(uri: vscode.Uri, message: string = ""): Promise<void> {
+    async downloadCopybooks(copybooks: string[], programName: string): Promise<void> {
+        // TODO do it right
```

Comment:
What means that? Can we remove it?

Reply:
Not yet. As I mentioned in the description, work is in progress, and this class will be also refactored.

Reply:
👍